src/python/pants/backend/codegen/protobuf/python/python_protobuf_module_mapper.py

```diff
@@ -33,29 +36,35 @@ class PythonProtobufMappingMarker(FirstPartyPythonMappingImplMarker):
 @rule(desc="Creating map of Protobuf targets to generated Python modules", level=LogLevel.DEBUG)
 async def map_protobuf_to_python_modules(
     protobuf_targets: AllProtobufTargets,
+    python_setup: PythonSetup,
     _: PythonProtobufMappingMarker,
 ) -> FirstPartyPythonMappingImpl:
+
     stripped_file_per_target = await MultiGet(
         Get(StrippedFileName, StrippedFileNameRequest(tgt[ProtobufSourceField].file_path))
         for tgt in protobuf_targets
     )
 
-    modules_to_providers: DefaultDict[str, list[ModuleProvider]] = defaultdict(list)
+    resolves_to_modules_to_providers: dict[ResolveName, DefaultDict[str, list[ModuleProvider]]] = {}
```

Comment:
better: `resolves_to_modules_to_providers: DefaultDict[ResolveName, DefaultDict[str, list[ModuleProvider]]] = defaultdict(lambda: defaultdict(list))`

Reply:
Ah that's a neat trick! Thanks!